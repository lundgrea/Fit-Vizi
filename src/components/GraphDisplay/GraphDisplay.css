.graph {
  font-size: 45px;
  background: white;
}

article {
  display: relative;
}