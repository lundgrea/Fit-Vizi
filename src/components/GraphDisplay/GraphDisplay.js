import React, { Component } from 'react';
import { Line } from 'react-chartjs-2';
import './GraphDisplay.css';

export class GraphDisplay extends Component {
  constructor(props) {
    super(props)
    this.state = {
      chartData: {
        labels: this.generateTimeArray(),
        datasets: [
          {
            label: "Power",
            data: this.generatePowerArray()
          }
        ]
      }
    }


  }

  generateTimeArray = () => {
    let samples = this.props.workoutData.samples
    let labels = samples.reduce((endValue, sample) => {
      endValue.push(sample.millisecondOffset)
      return endValue
    }, [])
    return labels
  }


  generatePowerArray = () => {
    let samples = this.props.workoutData.samples
    let data = samples.reduce((endValue, sample) => {
      endValue.push(sample.values.power)
      return endValue
    }, [])
    return data
  }

  render = () => {
    return (
      <section className='graph'>
        <h2>graph</h2>
        <article>
        <Line
          options={{
            responsive: true
          }}
          data={this.state.chartData}
        />
        </article>
      </section>
    )
  }

} 

export default GraphDisplay

