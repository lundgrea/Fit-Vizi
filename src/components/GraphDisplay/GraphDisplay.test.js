

import React from 'react';
import { shallow } from 'enzyme';
import { GraphDisplay } from './GraphDisplay';

describe('GraphDisplay', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<GraphDisplay 
      workoutData={[]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});