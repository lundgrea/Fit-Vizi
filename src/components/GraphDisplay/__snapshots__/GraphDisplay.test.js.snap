// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphDisplay should match the snapshot with all data passed in correctly 1`] = `
<section>
  <h2>
    Power Output Over Time (seconds)
  </h2>
  <article
    className="graph"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "data": Array [],
              "label": "Power Output",
            },
          ],
          "labels": Array [],
        }
      }
      options={
        Object {
          "onClick": [Function],
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Time (seconds)",
                },
              },
            ],
            "yAxes": Array [
              Object {
                "scaleLabel": Object {
                  "display": true,
                  "labelString": "Power Output",
                },
              },
            ],
          },
        }
      }
    />
  </article>
</section>
`;
